LayoutKit/Layouts/SizeLayout.swift

```diff
@@ -88,28 +151,90 @@ public class SizeLayout<V: View>: BaseLayout<V>, Layout {
-            if width == nil {
-                constrainedSize.width = subsize.width
-            }
-            if height == nil {
-                constrainedSize.height = subsize.height
-            }
+    public convenience init(maxSize: CGSize,
+                            alignment: Alignment? = nil,
+                            flexibility: Flexibility? = nil,
+                            viewReuseId: String? = nil,
+                            sublayout: Layout? = nil,
+                            config: (V -> Void)? = nil) {
+
+        self.init(maxWidth: maxSize.width,
+                  maxHeight: maxSize.height,
+                  alignment: alignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  sublayout: sublayout,
+                  config: config)
+    }
+
+    public convenience init(minSize: CGSize,
+                            alignment: Alignment? = nil,
+                            flexibility: Flexibility? = nil,
+                            viewReuseId: String? = nil,
+                            sublayout: Layout? = nil,
+                            config: (V -> Void)? = nil) {
+
+        self.init(minWidth: minSize.width,
+                  minHeight: minSize.height,
+                  alignment: alignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  sublayout: sublayout,
+                  config: config)
+    }
+
+    private static func defaultAlignment(maxWidth maxWidth: CGFloat?, maxHeight: CGFloat?) -> Alignment {
+        return Alignment(vertical: maxHeight == nil ? .fill : .center,
+                         horizontal: maxWidth == nil ? .fill : .center)
+    }
+
+    private static func defaultFlexibility(minWidth minWidth: CGFloat?,
+                                                    maxWidth: CGFloat?,
+                                                    minHeight: CGFloat?,
+                                                    maxHeight: CGFloat?) -> Flexibility {
+        let horizontal = dimensionFlex(min: minWidth, max: maxWidth)
+        let vertical = dimensionFlex(min: minHeight, max: maxHeight)
+        return Flexibility(horizontal: horizontal, vertical: vertical)
+    }
+
+    /// If we have an exact constraint on a dimension then it is inflexible.
+    /// Otherwise, it has default flex.
+    private static func dimensionFlex(min min: CGFloat?, max: CGFloat?) -> Flexibility.Flex {
+        return equals(min, max) ? Flexibility.inflexibleFlex : Flexibility.defaultFlex
+    }
+
+    /// Returns true if left and right are equal within a small tolerance.
+    /// A nil value is not equal to anything else.
+    private static func equals(left: CGFloat?, _ right: CGFloat?) -> Bool {
+        guard let left = left, right = right else {
+            // treat nil != nil
+            return false
         }
+        return abs(left - right) < 0.0001
```

Comment:
Will create static computed property. Can't currently create a static let because "Static stored properties not yet supported in generic types."